From c79c3556471bc0241b58a6dc07bea552b6820160 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 4 Mar 2015 21:05:26 +0200
Subject: [PATCH 0132/1240] fix: timer: initialize global variable

	- Initialize global variable with -1 to use it before relocation
	- Cannot init the global with 0, because there's conflict between
	  BSS section and relocation section.
	- When init the global with -1, the global will not be in the BSS
	  section

Change-Id: I8435ea2ec270fbcb20b6cbe45fcb662ac7d63020
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/17269
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/cpu/mvebu-common/timer_mvebu.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/cpu/mvebu-common/timer_mvebu.c b/arch/arm/cpu/mvebu-common/timer_mvebu.c
index e99b382..c2dba9e 100644
--- a/arch/arm/cpu/mvebu-common/timer_mvebu.c
+++ b/arch/arm/cpu/mvebu-common/timer_mvebu.c
@@ -45,7 +45,7 @@
 static ulong timestamp;
 static ulong lastdec;
 
-int timer_init_done;
+int timer_init_done = -1;
 
 int read_timer(void)
 {
@@ -56,7 +56,7 @@ int timer_init(void)
 {
 	unsigned int ctrl;
 
-	if (timer_init_done)
+	if (timer_init_done == 1)
 		return 0;
 
 	/* init the counter */
@@ -129,8 +129,8 @@ void __udelay(unsigned long usec)
 	uint current;
 	ulong delayticks;
 
-	/* In case udelay is called before timier was initialized */
-	if (!timer_init_done)
+	/* In case udelay is called before timer was initialized */
+	if (timer_init_done == -1)
 		timer_init();
 
 	delayticks = (usec * (MV_BOARD_REFCLK / 1000000));
-- 
1.9.1

